felix/bpf-gpl/types.h

```diff
@@ -120,22 +120,27 @@ struct cali_tc_ctx {
   struct fwd fwd;
 };
 
-static CALI_BPF_INLINE struct ethhdr* tc_ethhdr(struct cali_tc_ctx *ctx)
+static CALI_BPF_INLINE struct iphdr* ipv4hdr(struct cali_tc_ctx *ctx)
+{
+	return (struct iphdr *)ctx->ip_header;
+}
+
+static CALI_BPF_INLINE struct ethhdr* ethhdr(struct cali_tc_ctx *ctx)
 {
 	return (struct ethhdr *)ctx->data_start;
 }
 
-static CALI_BPF_INLINE struct tcphdr* tc_tcphdr(struct cali_tc_ctx *ctx)
+static CALI_BPF_INLINE struct tcphdr* tcphdr(struct cali_tc_ctx *ctx)
```

Comment:
i wonder whether it is a good call to remove the `tc_` prefix. After all they operate on a `tc` specific structure only. Although I always prefer shorter :sweat_smile: